File: erl2c/erl2ocaml_examples/mod04.ml
let rec mk_unit_remote'0 = function () -> Mod03.Unit0

let rec mk_left_remote'1 = function v_A -> Mod03.Left v_A

let rec mk_right_remote'1 = function v_B -> Mod03.Right v_B

let rec zero_remote'2 = function
  | v_U, v_Val -> ( match v_U with Mod03.Unit0 -> v_Val )

let rec un_pair_remote'1 = function
  | v_Pair -> ( match v_Pair with Mod03.Pair (v_A, v_B) -> (v_A, v_B) )

File: erl2c/erl2ocaml_examples/mod04.erl
-module(mod04).
-compile(export_all).

-spec mk_unit_remote() -> mod03:unit0().
mk_unit_remote() -> mod03:unit0{}.

-spec mk_left_remote(A) -> mod03:either(A, _).
mk_left_remote(A) -> mod03:left{A}.

-spec mk_right_remote(B) -> mod03:either(_, B).
mk_right_remote(B) -> mod03:right{B}.
